core/translate/emitter.rs

```diff
@@ -822,6 +839,84 @@ fn emit_update_insns(
             }
         }
     }
+
+    for (index, (idx_cursor_id, record_reg)) in plan.indexes_to_update.iter().zip(&index_cursors) {
+        if index.unique {
```

Comment:
i'd rather just 

```rust
if !index.unique {
  continue;
}
```

to flatten the indentation. right now the if block is so long that it's not immediately obvious that nothing is done for non-unique indexes inside this loop